torchtune/utils/collate.py

```diff
@@ -71,3 +71,20 @@ def padded_collate(
             value=padding_idx,
         )
     return input_ids, labels
+
+
+def padded_collate_dpo(
+    batch: List[TokenPair],
+    padding_idx: int = 0,
+    ignore_idx: int = -100,
+) -> Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
Could you add docstrings and also explain why this is needed for DPO instead of the normal `padded_collate`?

Reply:
I've documented the padded_collate_dpo function.
And, it seems that the input data formats for padded_collate and padded_collate_dpo are different – one expects a list of token pairs, while the other expects a list of dictionaries. Therefore, using padded_collate for the data formatted as dictionaries might not be suitable.